cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -74,6 +74,7 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
       repoRequest.getRepository().validate();
     } catch (InvalidRepositoryConfigException e) {
       responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson(new SetRepositoryResponse(e)));
+      return;
```

Comment:
Do we have a unit test covering this code path ?

Reply:
Added the tests